sdk/core/core/src/az_http_response_parser.c

```diff
@@ -298,3 +298,32 @@ az_http_response_get_body(az_span const self, az_pair * const p_last_header, az_
   AZ_RETURN_IF_FAILED(az_http_response_parser_get_body(&parser, body));
   return AZ_OK;
 }
+
+/**
+ * Get an HTTP header by name.
+ *
+ * Returns @AZ_ERROR_HTTP_NO_MORE_ITEMS
+ */
+AZ_NODISCARD az_result az_http_response_get_header_by_name(
+    az_span const self,
+    az_span const header_name,
+    az_span * const header_value) {
+  AZ_CONTRACT_ARG_NOT_NULL(header_value);
+
+  az_http_response_parser parser;
+  AZ_RETURN_IF_FAILED(az_http_response_parser_init(&parser, self));
```

Comment:
so, it's kind of weird that this API is calling `init` and the other APIs aren't (like `az_span_reader_get_http_status_line`). How do we know when we need to fist call init and when the API will do it for us?

Reply:
`az_span_reader_get_http_status_line` is an internal static function used in HTTP response parser.